include/imex/Dialect/Dist/IR/DistOps.td

```diff
@@ -46,48 +75,149 @@ def Dist_Dialect : Dialect {
+    let arguments = (ins AnyType:$team);
+    let results = (outs I64);
+    let builders = [
+        OpBuilder<(ins "::mlir::Value":$team), [{
+            build($_builder, $_state, $_builder.getI64Type(), team);
+        }]>,
+    ];
+}
 
-    // result is an Integer Id
+def PRankOp : Dist_Op<"prank", [Pure]> {
+    let summary = "Process rank in team";
+    let arguments = (ins AnyType:$team);
     let results = (outs I64);
+    let builders = [
+        OpBuilder<(ins "::mlir::Value":$team), [{
+            build($_builder, $_state, $_builder.getI64Type(), team);
+        }]>,
+    ];
 }
 
-// Get the offsets (one for each dimension) of the local partition of a distributed PTensor in number of elements.
-// Partitionings can be N-dimensional but must cut only the first N dimensions.
-def LocalOffsetsOp : Dist_Op<"local_offsets", []> {
-    // Id of tensor as returned by RegisterPTensorOp
-    let arguments = (ins I64Attr: $rank, I64: $ptensor);
+def DistInfoOp : Dist_Op<"distinfo", []> {
+    let summary = "Initialize distributed meta information about a new PTensor.";
+    let description = [{
+        The distributed meta information about a new PTensor provides
+          - the rank of the tensor
+          - the process-local shape
+          - the process-local offsets
+          - the distributed team
+    }];
+    // Global shape needed for initial registration. Views are handled by a separate op.
+    let arguments = (ins I64Attr:$rank, AnyType:$shape, AnyType:$team);
+    let results = (outs Dist_Info);
+}
+
+def InitDistTensorOp : Dist_Op<"init_dist_tensor", [Pure]> {
+    let summary = "Bind a PTensor to a DistInfo";
+    let arguments = (ins AnyType:$ptensor, Dist_Info:$info);
+    let results = (outs Dist_Tensor);
+    let builders = [
+      OpBuilder<(ins "::mlir::Value":$ptensor, "::mlir::Value":$info), [{
+            build($_builder, $_state,
+                  ::imex::dist::DistTensorType::get($_builder.getContext(), ptensor.getType().dyn_cast<::imex::ptensor::PTensorType>()),
+                  ptensor, info);
+      }]>,
+    ];
+}
 
-    // result is a 1d memref
+def GetPTensorOp : Dist_Op<"get_ptensor", [Pure]> {
+    let summary = "Get the PTensor from a given DistTensor.";
+    let arguments = (ins Dist_Tensor:$d_tensor);
     let results = (outs AnyType);
+    let builders = [
+      OpBuilder<(ins "::mlir::Value":$dtensor), [{
+            build($_builder, $_state,
+                  dtensor.getType().dyn_cast<::imex::dist::DistTensorType>().getPTensorType(), dtensor);
+      }]>,
+    ];
 }
 
-// Get the shape (one size for each dimension) of the local partition of a distributed PTensor in number of elements.
-// Partitionings can be N-dimensional but must cut only the first N dimensions.
-def LocalShapeOp : Dist_Op<"local_shape", []> {
-    // Id of tensor as returned by RegisterPTensorOp
-    let arguments = (ins I64Attr: $rank, I64: $ptensor);
+def GetInfoOp : Dist_Op<"get_info", [Pure]> {
```

Comment:
What is the difference between `GetInfoOp` and `DistInfoOp`?

Reply:
`DistInfoOP` is the initialization/creation of an `DistInfo` from tensor + shape + team. `GetInfoOp` returns the `DistInfo` from a given `DistTensor`.